portalnet/src/overlay_service.rs

```diff
@@ -2512,6 +2512,25 @@ pub fn propagate_gossip_cross_thread<TContentKey: OverlayContentKey>(
         })
         .collect();
 
+    if all_nodes.is_empty() {
+        // If there are no connected nodes, use the nodes regardless of connectivity.
+        warn!("No connected nodes, using disconnected nodes for gossip.");
```

Comment:
I think it's okay to drop the comment, since the log explains the same situation adequately.